src/time/windows/SDL_systime.c

```diff
@@ -0,0 +1,139 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2024 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+#include "SDL_internal.h"
+
+#ifdef SDL_TIME_WINDOWS
+
+#include "../../core/windows/SDL_windows.h"
+#include <minwinbase.h>
+#include <timezoneapi.h>
+
+#include "../SDL_time_c.h"
+
+#define NS_PER_WINDOWS_TICK   100ULL
+#define WINDOWS_TICK          10000000ULL
+#define UNIX_EPOCH_OFFSET_SEC 11644473600ULL
+
+void SDL_GetSystemTimeLocalePreferences(SDL_DATE_FORMAT *df, SDL_TIME_FORMAT *tf)
+{
+    WCHAR str[80]; /* Per the docs, the time and short date format strings can be a max of 80 characters. */
+
+    if (GetLocaleInfoW(LOCALE_USER_DEFAULT, LOCALE_SSHORTDATE, str, sizeof(str) / sizeof(WCHAR))) {
+        LPWSTR s = str;
+        while (*s) {
+            switch (*s++) {
+            case L'y':
+                *df = SDL_DATE_FORMAT_YYYYMMDD;
+                goto found_date;
+            case L'd':
+                *df = SDL_DATE_FORMAT_DDMMYYYY;
+                goto found_date;
+            case L'M':
+                *df = SDL_DATE_FORMAT_MMDDYYYY;
+                goto found_date;
+            default:
+                break;
+            }
+        }
+    }
+
+found_date:
+
+    /* Figure out the preferred system date format. */
+    if (GetLocaleInfoW(LOCALE_USER_DEFAULT, LOCALE_STIMEFORMAT, str, sizeof(str) / sizeof(WCHAR))) {
+        LPWSTR s = str;
+        while (*s) {
+            switch (*s++) {
+            case L'H':
+                *tf = SDL_TIME_FORMAT_24HR;
+                return;
+            case L'h':
+                *tf = SDL_TIME_FORMAT_12HR;
+                return;
+            default:
+                break;
+            }
+        }
+    }
+}
+
+int SDL_GetCurrentTime(SDL_Time *ticks)
+{
+    FILETIME ft;
+
+    if (!ticks) {
+        return SDL_SetError("Invalid parameter");
```

Comment:
This can be fixed after the merge, but FYI, this would normally be:
```c
return SDL_InvalidParamError("ticks");
```

Reply:
It's fixed now.